### PR TITLE
Adds rules for foo: A() and foo: EmberObject.create

### PR DESCRIPTION
**Pull request by ntgussoni**
#300 

Adds warnings for: 
 - someArray: A()
 - someArray: Ember.A()
 - someArray: EmberAlias.A()
 - someProp: Object()
 - someProp: EmberObject.create({...})

### CODE REVIEW

File: lib/rules/avoid-leaking-state-in-ember-objects.js [hunk 1]
```diff
@@ -52,6 +38,47 @@ module.exports = {
...
+      return ember.isFunctionExpression(value) ||
+        utils.isLiteral(value) ||
+        utils.isIdentifier(value) ||
+        (utils.isCallExpression(value) && !isEmberArrayCall(value) && !isEmberObjectCall(value)) ||
+        utils.isBinaryExpression(value) ||
+        utils.isTemplateLiteral(value) ||
+        utils.isTaggedTemplateExpression(value) ||
+        utils.isMemberExpression(value) ||
+        utils.isUnaryExpression(value);
+    };
```
Comment (Turbo87) [inline lib/rules/avoid-leaking-state-in-ember-objects.js]: why did you move this into the function? this will redefine the function over and over for each call. since the function is not using any variables from the outer function I don't see a reason for moving it 🤔

Comment (ntgussoni) [reply lib/rules/avoid-leaking-state-in-ember-objects.js]: You are actually right! I only need emberImportAliasName which i can pass as an argument. 

For the rest of the code, i wonder if there's a better way of doing the checks. 

Thx for the feedback!

**Comment by rwjblue** [comment]
@ntgussoni Had a chance to pick this back up again?

**Comment by ntgussoni** [comment]
> @ntgussoni Had a chance to pick this back up again?

@rwjblue 
I've been completely consumed by work these couple of weeks, but I'll try to do the requested changes tomorrow. 
Thx!

**Comment by ntgussoni** [comment]
@Turbo87 @rwjblue  I made the requested changes. Let me know if you think we can make further improvements.

File: lib/rules/avoid-leaking-state-in-ember-objects.js [hunk 2]
```diff
@@ -60,6 +87,12 @@ module.exports = {
     };
 
     return {
+      ImportDeclaration(node) {
+        if (!emberImportAliasName) {
+          emberImportAliasName = ember.getEmberImportAliasName(node);
```
Comment (rwjblue) [inline lib/rules/avoid-leaking-state-in-ember-objects.js]: The idea that there is only one import is flawed, we should account for something like:

```js
import Ember from 'ember';
import Em from 'ember';

export default Ember.Object.extend({
  foo: Em.A(),
})
```

Can you create a "bad" test for this?

---

Additionally, storing the value in module scope means that this is not isolated per-file. Specifically, running two different files one with `import Ember from 'ember'` and the other with `import Em from 'ember'` would not work.

Comment (ntgussoni) [reply lib/rules/avoid-leaking-state-in-ember-objects.js]: Sure thing! I can create a bad test for this.
Didn't really think that it would be shared across files. 

Thanks for you feedback! I'll make the changes as soon as I can

Comment (ntgussoni) [reply lib/rules/avoid-leaking-state-in-ember-objects.js]: Hey @rwjblue  I see that it's often assumed (in other rules) that there's only one ember import. What should we do about that. Im not entirely sure how to do a "bad/invalid" test for that matter because I think that i should write a valid one to check this case.
